lib/spack/spack/installer.py

```diff
@@ -763,6 +763,13 @@ def _add_bootstrap_compilers(self, compiler, architecture, pkgs, request, all_de
         for (comp_pkg, is_compiler) in packages:
             if package_id(comp_pkg) not in self.build_tasks:
                 self._add_init_task(comp_pkg, request, is_compiler, all_deps)
+            elif is_compiler:
+                # Requeue ensuring we treat it as a compiler
```

Comment:
This isn't really re-queueing the task (as in the work done in `_push_task`), but modifying existing tasks in place.